In [1]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt 
import seaborn as sns
import plotly.express as px
import faculty.datasets as datasets
from pandas.api.types import CategoricalDtype
import matplotlib.ticker as mtick
import statsmodels.api as sm


from pandas import Series, DataFrame
import os
from functools import reduce
vcvtype = 'nonrobust'

In [2]:
temdifference = pd.read_csv('/project/temdifference.csv')
all_football_18 = pd.read_csv('/project/all_football_18.csv')
market_value_all = pd.read_csv('/project/market_value_all.csv')
temdifference
market_value_all

Unnamed: 0       Country  AvgTemperature  TemperatureDifference
0            0     Argentina        9.730419              -8.818761
1            1     Australia       11.830601              -6.718579
2            2       Belgium       19.479053               0.929872
3            3        Brazil       19.686096               1.136916
4            4      Colombia       13.899818              -4.649362
5            5    Costa Rica       22.836976               4.287796
6            6       Croatia       22.609290               4.060109
7            7       Denmark       18.753188               0.204007
8            8         Egypt       30.083789              11.534608
9            9        France       13.907559              -4.641621
10          10       Germany       18.722222               0.173042
11          11       Iceland        9.663934              -8.885246
12          12         Japan       23.262599               4.713418
13          13        Mexico       20.540680               1.991500
14          14       Morocco       20.108379               1.559199
15          15       Nigeria       28.053279               9.504098
16          16        Panama       26.645719               8.096539
17          17        Poland       20.906193               2.357013
18          18      Portugal       19.738616               1.189435
19          19        Russia       18.549180               0.000000
20          20  Saudi Arabia       37.094262              18.545082
21          21       Senegal       26.004554               7.455373
22          22   South Korea       24.826047               6.276867
23          23         Spain       22.928051               4.378871
24          24        Sweden       19.315118               0.765938
25          25   Switzerland       20.065877               1.516697
26          26       Tunisia       26.534608               7.985428
27          27       England       18.157559              -0.391621
28          28       Uruguay       10.268670              -8.280510
29          29          Peru       19.976250               1.427070
30          30          Iran       20.541000               1.991820
31          31        Serbia       12.843625              -5.705555

Unnamed: 0       country confederation  squad_size  avg_age  \
0            0     Argentina      CONMEBOL          36     25.9   
1            1     Australia           AFC          23     25.6   
2            2       Belgium          UEFA          23     27.8   
3            3        Brazil      CONMEBOL          23     26.5   
4            4      Colombia      CONMEBOL          26     28.5   
5            5    Costa Rica      CONCACAF          23     26.4   
6            6       Croatia          UEFA          26     27.1   
7            7       Denmark          UEFA          30     26.5   
8            8       England          UEFA          25     25.4   
9            9        France          UEFA          24     27.3   
10          10       Germany          UEFA          21     27.5   
11          11          Iran           AFC          27     27.4   
12          12       Iceland          UEFA          23     28.7   
13          13         Japan           AFC          28     27.7   
14          14        Mexico      CONCACAF          23     28.7   
15          15       Morocco           CAF          25     26.5   
16          16       Nigeria           CAF          24     26.8   
17          17          Peru      CONMEBOL          27     29.4   
18          18        Poland          UEFA          26     27.3   
19          19      Portugal          UEFA          23     27.3   
20          20        Russia          UEFA          25     27.3   
21          21  Saudi Arabia           AFC          23     25.2   
22          22       Senegal           CAF          26     27.3   
23          23        Serbia          UEFA          28     26.7   
24          24   South Korea           AFC          25     27.2   
25          25         Spain          UEFA          25     27.3   
26          26        Sweden          UEFA          26     28.2   
27          27   Switzerland          UEFA          22     26.8   
28          28       Tunisia           CAF          33     26.2   
29          29       Uruguay      CONMEBOL          27     27.4   
30          30         Egypt           CAF          27     30.3   
31          31        Panama      CONCACAF          23     28.4   

   market_value_in_euros  market_value_digits  \
0         747.50 million                  747   
1         32.85 million                    32   
2         419.00 million                  419   
3         870.50 million                  870   
4         282.95 million                  282   
5         24.25 million                    24   
6         317.00 million                  317   
7         289.30 million                  289   
8           1.23 billion                 1000   
9         978.80 million                  978   
10        572.00 million                  572   
11        61.13 million                    61   
12        23.25 million                    23   
13        118.10 million                  118   
14        214.00 million                  214   
15        217.70 million                  217   
16        277.55 million                  277   
17         67.30 million                   67   
18        292.80 million                  292   
19        775.00 million                  775   
20        153.25 million                  153   
21        26.25 million                    26   
22        348.80 million                  348   
23        344.00 million                  344   
24        127.95 million                  127   
25        728.00 million                  728   
26        203.40 million                  203   
27        189.20 million                  189   
28        39.18 million                    39   
29        263.20 million                  263   
30        19.70 million                    19   
31         10.17 million                   10   

    avg_market_value_of_players_in_mill_euros  
0                                       20.75  
1                                        1.39  
2                             

In [3]:
# The final position of country teams should be included in dataset to act as the dependent variable
temdifference = temdifference.join(all_football_18['Final_positional_ranking'])
temdifference

Unnamed: 0       Country  AvgTemperature  TemperatureDifference  \
0            0     Argentina        9.730419              -8.818761   
1            1     Australia       11.830601              -6.718579   
2            2       Belgium       19.479053               0.929872   
3            3        Brazil       19.686096               1.136916   
4            4      Colombia       13.899818              -4.649362   
5            5    Costa Rica       22.836976               4.287796   
6            6       Croatia       22.609290               4.060109   
7            7       Denmark       18.753188               0.204007   
8            8         Egypt       30.083789              11.534608   
9            9        France       13.907559              -4.641621   
10          10       Germany       18.722222               0.173042   
11          11       Iceland        9.663934              -8.885246   
12          12         Japan       23.262599               4.713418   
13          13        Mexico       20.540680               1.991500   
14          14       Morocco       20.108379               1.559199   
15          15       Nigeria       28.053279               9.504098   
16          16        Panama       26.645719               8.096539   
17          17        Poland       20.906193               2.357013   
18          18      Portugal       19.738616               1.189435   
19          19        Russia       18.549180               0.000000   
20          20  Saudi Arabia       37.094262              18.545082   
21          21       Senegal       26.004554               7.455373   
22          22   South Korea       24.826047               6.276867   
23          23         Spain       22.928051               4.378871   
24          24        Sweden       19.315118               0.765938   
25          25   Switzerland       20.065877               1.516697   
26          26       Tunisia       26.534608               7.985428   
27          27       England       18.157559              -0.391621   
28          28       Uruguay       10.268670              -8.280510   
29          29          Peru       19.976250               1.427070   
30          30          Iran       20.541000               1.991820   
31          31        Serbia       12.843625              -5.705555   

    Final_positional_ranking  
0                         16  
1                         25  
2                          3  
3                          5  
4                         10  
5                         30  
6                          2  
7                         13  
8                         31  
9                          4  
10                         1  
11                        23  
12                        27  
13                        19  
14                        15  
15                         9  
16                        24  
17                        22  
18                        32  
19                        21  
20                        26  
21                        14  
22                         7  
23                        29  
24                        17  
25                        18  
26                        20  
27                        12  
28                         8  
29                        11  
30                        28  
31                         6

In [4]:
market_value_all = market_value_all.rename(columns = {"country":"Country"})
market_value_all
regression = pd.merge(temdifference, market_value_all, on = "Country", how = "left")
regression

Unnamed: 0       Country confederation  squad_size  avg_age  \
0            0     Argentina      CONMEBOL          36     25.9   
1            1     Australia           AFC          23     25.6   
2            2       Belgium          UEFA          23     27.8   
3            3        Brazil      CONMEBOL          23     26.5   
4            4      Colombia      CONMEBOL          26     28.5   
5            5    Costa Rica      CONCACAF          23     26.4   
6            6       Croatia          UEFA          26     27.1   
7            7       Denmark          UEFA          30     26.5   
8            8       England          UEFA          25     25.4   
9            9        France          UEFA          24     27.3   
10          10       Germany          UEFA          21     27.5   
11          11          Iran           AFC          27     27.4   
12          12       Iceland          UEFA          23     28.7   
13          13         Japan           AFC          28     27.7   
14          14        Mexico      CONCACAF          23     28.7   
15          15       Morocco           CAF          25     26.5   
16          16       Nigeria           CAF          24     26.8   
17          17          Peru      CONMEBOL          27     29.4   
18          18        Poland          UEFA          26     27.3   
19          19      Portugal          UEFA          23     27.3   
20          20        Russia          UEFA          25     27.3   
21          21  Saudi Arabia           AFC          23     25.2   
22          22       Senegal           CAF          26     27.3   
23          23        Serbia          UEFA          28     26.7   
24          24   South Korea           AFC          25     27.2   
25          25         Spain          UEFA          25     27.3   
26          26        Sweden          UEFA          26     28.2   
27          27   Switzerland          UEFA          22     26.8   
28          28       Tunisia           CAF          33     26.2   
29          29       Uruguay      CONMEBOL          27     27.4   
30          30         Egypt           CAF          27     30.3   
31          31        Panama      CONCACAF          23     28.4   

   market_value_in_euros  market_value_digits  \
0         747.50 million                  747   
1         32.85 million                    32   
2         419.00 million                  419   
3         870.50 million                  870   
4         282.95 million                  282   
5         24.25 million                    24   
6         317.00 million                  317   
7         289.30 million                  289   
8           1.23 billion                 1000   
9         978.80 million                  978   
10        572.00 million                  572   
11        61.13 million                    61   
12        23.25 million                    23   
13        118.10 million                  118   
14        214.00 million                  214   
15        217.70 million                  217   
16        277.55 million                  277   
17         67.30 million                   67   
18        292.80 million                  292   
19        775.00 million                  775   
20        153.25 million                  153   
21        26.25 million                    26   
22        348.80 million                  348   
23        344.00 million                  344   
24        127.95 million                  127   
25        728.00 million                  728   
26        203.40 million                  203   
27        189.20 million                  189   
28        39.18 million                    39   
29        263.20 million                  263   
30        19.70 million                    19   
31         10.17 million                   10   

    avg_market_value_of_players_in_mill_euros  
0                                       20.75  
1                                        1.39  
2                             

Unnamed: 0_x       Country  AvgTemperature  TemperatureDifference  \
0              0     Argentina        9.730419              -8.818761   
1              1     Australia       11.830601              -6.718579   
2              2       Belgium       19.479053               0.929872   
3              3        Brazil       19.686096               1.136916   
4              4      Colombia       13.899818              -4.649362   
5              5    Costa Rica       22.836976               4.287796   
6              6       Croatia       22.609290               4.060109   
7              7       Denmark       18.753188               0.204007   
8              8         Egypt       30.083789              11.534608   
9              9        France       13.907559              -4.641621   
10            10       Germany       18.722222               0.173042   
11            11       Iceland        9.663934              -8.885246   
12            12         Japan       23.262599               4.713418   
13            13        Mexico       20.540680               1.991500   
14            14       Morocco       20.108379               1.559199   
15            15       Nigeria       28.053279               9.504098   
16            16        Panama       26.645719               8.096539   
17            17        Poland       20.906193               2.357013   
18            18      Portugal       19.738616               1.189435   
19            19        Russia       18.549180               0.000000   
20            20  Saudi Arabia       37.094262              18.545082   
21            21       Senegal       26.004554               7.455373   
22            22   South Korea       24.826047               6.276867   
23            23         Spain       22.928051               4.378871   
24            24        Sweden       19.315118               0.765938   
25            25   Switzerland       20.065877               1.516697   
26            26       Tunisia       26.534608               7.985428   
27            27       England       18.157559              -0.391621   
28            28       Uruguay       10.268670              -8.280510   
29            29          Peru       19.976250               1.427070   
30            30          Iran       20.541000               1.991820   
31            31        Serbia       12.843625              -5.705555   

    Final_positional_ranking  Unnamed: 0_y confederation  squad_size  avg_age  \
0                         16             0      CONMEBOL          36     25.9   
1                         25             1           AFC          23     25.6   
2                          3             2          UEFA          23     27.8   
3                          5             3      CONMEBOL          23     26.5   
4                         10             4      CONMEBOL          26     28.5   
5                         30             5      CONCACAF          23     26.4   
6                          2             6          UEFA          26     27.1   
7                         13             7          UEFA          30     26.5   
8                         31            30           CAF          27     30.3   
9                          4             9          UEFA          24     27.3   
10                         1            10          UEFA          21     27.5   
11                        23            12          UEFA          23     28.7   
12                        27            13           AFC          28     27.7   
13                        19            14      CONCACAF          23     28.7   
14                        15            15           CAF          25     26.5   
15                         9            16           CAF          24     26.8   
16                        24            31      CONCACAF          23     28.4   
17                        22            18          UEFA          26     27.3   
18                        32            19          UEF

In [6]:
X = regression[["TemperatureDifference", "avg_age", "avg_market_value_of_players_in_mill_euros"]] # independent variables
X = sm.add_constant(X)
Y = regression["Final_positional_ranking"] # dependent variable
model = sm.OLS(Y,X)
results = model.fit(cov_type=vcvtype)
print(results.summary())

                               OLS Regression Results                               
Dep. Variable:     Final_positional_ranking   R-squared:                       0.184
Model:                                  OLS   Adj. R-squared:                  0.096
Method:                       Least Squares   F-statistic:                     2.098
Date:                      Mon, 13 Dec 2021   Prob (F-statistic):              0.123
Time:                              11:57:09   Log-Likelihood:                -113.29
No. Observations:                        32   AIC:                             234.6
Df Residuals:                            28   BIC:                             240.4
Df Model:                                 3                                         
Covariance Type:                  nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------

It can be seen that only average market value of players in each country team is statistically significant at 10% significance level. Holding anything else constant, increasing 1 million euros in average market value of team players would move the rank forward by 0.2499 on average.

It can be found that R-squared is quite low as well (i.e. 0.184) which means those independent variables put in the model did not explain the variation of the dependent variable (i.e. final ranking) very well. Also, the number of observations are for all variables are only 32, which is quite small.